sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -34,3 +39,44 @@ case class GpuMd5(child: Expression)
     }
   }
 }
+
+case class GpuMurmur3Hash(child: Seq[Expression]) extends GpuExpression {
+  override def dataType: DataType = IntegerType
+
+  override def toString: String = s"hash($child)"
+  def nullable: Boolean = children.exists(_.nullable)
+  def children: Seq[Expression] = child
+
+  def columnarEval(batch: ColumnarBatch): Any = {
+    var nullStrScalar: Scalar = null
+    val rows = batch.numRows()
+    val childEvals: ArrayBuffer[Any] = new ArrayBuffer[Any](children.length)
```

Comment:
`childEvals` is populated before `columns` is, but `columns` is the only one that is closed at the end. If we got an exception while processing we would leak anything in `childEvals` that is not in `columns` yet.

I think it would be cleaner to just not have childEvals and loop through children instead so we avid the problem entirely.

```
children.foreach { expr =>
    expr.columnarEval(batch) match { ...
    }
}
```